File: 프로그래머스/2/42586. 기능개발/기능개발.java
class Solution {
    public int[] solution(int[] progresses, int[] speeds) {
        
        int[] temp = new int[100];
        int day = 0;
        
        for(int i=0; i<progresses.length; i++){
            while(progresses[i] + (speeds[i] * day) < 100){
                day++;
            }
            temp[day]++;
        }
        
        int count = 0;
        
        for(int n: temp){
            if(n != 0){
                count++;
            }
        }
        
        int[] answer = new int[count]; 
        
        count = 0;
        
        for(int n:temp){
            if(n != 0){
                answer[count++] = n;
            }
        }
        return answer;
    }
}

File: 프로그래머스/2/42586. 기능개발/기능개발.js
function solution(progresses, speeds) {
   let takeDays = [];

    for(let i=0; i<progresses.length; i++) {
      const leftProgresses = 100-progresses[i]
      takeDays.push(Math.ceil(leftProgresses/speeds[i]))
    }

    let compare = takeDays[0]; 
    let count = 0;
    let result = [];

    for(let i = 0; i < takeDays.length; i++) {
        if(compare >= takeDays[i]) {
            count++;
        } else {
            result.push(count);
            count = 1;
            compare = takeDays[i];
        }
    }

    result.push(count);

    return result;
}

